wordpress-mobile/WordPress-iOS
# Site Settings: Timezone

Fixes #7831 

This PR adds support for changing Timezone settings in Site settings.
This PR is one of the first in a line of PRs which will further enhance site settings, by bringing more settings to mobile.

**Notes:**
- We took a call to not save the Timezones API to a local.json file.
  Instead the way it's implemented is that the app will make an API call the first time, when there is no Timezone data in the core data DB, to get the data and save it to DB.
  And subsequent views of the screen will just load the data from DB.
  Reasoning to not go with the local.json file:
  - While the Timezones API would probably change very infrequently, the possibility still exists and making a one time API call should be fine, as otherwise we'd have to keep a check on the Timezone API every once in a while and make sure it hasn't changed, if it did change then we'd have to ship out an app update.
- Also it seems that you cannot see the timezone settings for Jetpack enabled sites in the Calypso editor.
  You have to go to `wp-admin/options-general.php` and then you can see the time zone option there.

**To test:**
- Go to site settings for a Jetpack enabled WordPress site on mobile as well as web.
- Check if the timezone displayed on mobile is the same as displayed on web.
- Try changing the time zone to another time zone
- Also try experimenting around changing to the manual offset section as well
- Observe the same is reflected on web as well.
- Repeat the above steps for a wordpress.com website as well
